src/memray/_memray/frame_tree.h

```diff
@@ -9,12 +9,23 @@ namespace memray::tracking_api {
 class FrameTree
 {
   public:
-    using index_t = uint32_t;
+    using index_t = uint32_t;  // TODO: Shouldn't this be size_t?
```

Comment:
OK. That's unrelated to this PR, just something I noticed while going through this one. Rather than try to address that here, I'll remove the `TODO` comment, and add an issue for fixing this - https://github.com/bloomberg/memray/issues/285